src/commands.rs

```diff
@@ -2042,93 +2044,118 @@ fn cmd_log(ui: &mut Ui, command: &CommandHelper, args: &LogArgs) -> Result<(), C
+                        &workspace_command,
+                        &commit,
+                        matcher.as_ref(),
+                        diff_format,
+                    )?;
+                }
+                let node_symbol = if is_checkout { b"@" } else { b"o" };
+                graph.add_node(
+                    &index_entry.position(),
+                    &graphlog_edges,
+                    node_symbol,
+                    &buffer,
                 )?;
             }
-            let node_symbol = if is_checkout { b"@" } else { b"o" };
-            graph.add_node(
-                &index_entry.position(),
-                &graphlog_edges,
-                node_symbol,
-                &buffer,
-            )?;
-        }
-    } else {
-        let iter: Box<dyn Iterator<Item = IndexEntry>> = if args.reversed {
-            Box::new(revset.iter().reversed())
         } else {
-            Box::new(revset.iter())
-        };
-        for index_entry in iter {
-            let commit = store.get_commit(&index_entry.commit_id())?;
-            template.format(&commit, formatter)?;
-            if let Some(diff_format) = diff_format {
-                show_patch(
-                    formatter,
-                    &workspace_command,
-                    &commit,
-                    matcher.as_ref(),
-                    diff_format,
-                )?;
+            let iter: Box<dyn Iterator<Item = IndexEntry>> = if args.reversed {
+                Box::new(revset.iter().reversed())
+            } else {
+                Box::new(revset.iter())
+            };
+            for index_entry in iter {
+                let commit = store.get_commit(&index_entry.commit_id())?;
+                template.format(&commit, formatter)?;
+                if let Some(diff_format) = diff_format {
+                    show_patch(
+                        formatter,
+                        &workspace_command,
+                        &commit,
+                        matcher.as_ref(),
+                        diff_format,
+                    )?;
+                }
+            }
+        }
+    }
+
+    // Check to see if the user might have specified a path when they intended
+    // to specify a revset.
+    if let [only_path] = args.paths.as_slice() {
+        if only_path == "." {
+            // For users of e.g. Mercurial, where `.` indicates the current commit.
+            ui.write_warn(&format!(
+                "warning: The argument {only_path:?} is being interpreted as a path, but this is \
+                 often not useful because all non-empty commits touch '.'.  If you meant to show \
+                 the working copy commit, pass -r '@' instead."
+            ))?;
+        } else if revset.is_empty() {
+            if let Ok(path_as_revset) = workspace_command.resolve_revset(only_path) {
+                if !path_as_revset.is_empty() {
+                    ui.write_warn(&format!(
```

Comment:
Nit: I don't think `path_as_revset.is_empty()` is needed here. If the path can be parsed as a revset, it makes sense to suggest `-r<path>`.